src/app/static/src/app/components/forms/DynamicForm.vue

```diff
@@ -24,100 +30,85 @@
         id: string
     }
 
-    interface ValidationResult {
-        data: any
-        valid: boolean
-        missingValues: string[]
-    }
-
-    interface Data {
-        validated: boolean
-    }
-
     interface Methods {
-        validate: (formData: FormData) => ValidationResult
-        submit: (e: Event) => ValidationResult
+        buildValue: (control: DynamicControl) => string | string[] | number | null
+        submit: (e: Event) => DynamicFormData
+        change: () => void
     }
 
     interface Computed {
-        formControlLookup: Dict<Control>
+        controls: Control[]
+        disabled: boolean
     }
 
-    export default Vue.extend <Data, Methods, Computed, Props>({
-        name: "DynamicForm",
-        data() {
-            return {
-                validated: false
-            }
+    const props = {
+        id: {
+            type: String,
+            default: "d-form"
         },
-        props: {
-            id: {
-                type: String,
-                default: "d-form"
-            },
-            submitText: {
-                type: String,
-                default: "Submit"
-            },
-            includeSubmitButton: {
-                type: Boolean,
-                default: true
-            },
-            formMeta: {
-                type: Object
-            }
+        submitText: {
+            type: String,
+            default: "Submit"
+        },
+        includeSubmitButton: {
+            type: Boolean,
+            default: true
+        },
+        formMeta: {
+            type: Object
+        }
+    };
+
+    export default Vue.extend <{}, Methods, Computed, Props>({
+        name: "DynamicForm",
+        props: props,
+        model: {
+            prop: "formMeta",
+            event: "change"
         },
         components: {
             BForm,
             DynamicFormControlGroup,
             DynamicFormControlSection
         },
         computed: {
-            formControlLookup() {
-                const dict = {} as Dict<Control>;
+            controls() {
+                const controls: Control[] = [];
                 this.formMeta.controlSections.map(s => {
                     s.controlGroups.map(g => {
                         g.controls.map(c => {
-                            dict[c.name] = c
+                            controls.push(c);
                         })
                     })
                 });
-
-                return dict;
+                return controls;
+            },
+            disabled() {
+                return this.controls
+                    .filter(c => c.required && (c.value == null || c.value == ""))
+                    .length > 0
             }
         },
         methods: {
-            validate(formData: FormData) {
```

Comment:
Rather than "validating" on submit, which was actually just checking that required values as present, we now simply disable the submit button until required values are present. When the button is enabled and clicked, the form data is returned.